lib/vrl/tests/tests/expressions/arithmetic/union/multiple_maps.vrl

```diff
@@ -0,0 +1,3 @@
+#result: { "foo": 1, "bar": 2, "zub": 3, "zoob": 4 }
+
+{ "foo": 1, "bar": 99 } | { "bar": 2 } | { "zub": 3, "zoob": 99 } | { "zoob": 4 }
```

Comment:
Would be good to add a test to see what happens with deep merges.

Reply:
These merges are only ever shallow. If you want deep, you need the `merge` function. Should it be the other way round?

Reply:
No, I meant we should add a test that validates that the merges are indeed shallow 😄